### PR TITLE
added color

### LINKED ISSUE #5: Add colors to output
Even if they are not visible by everyone, colors help to read a diff.

It would be nice to add green and red background colors to the added and removed lines, in the output.

### PR DESCRIPTION
**Pull request by ademolaomosanya**
fixes #5 
this pr fixes Add colors to output #5

### CODE REVIEW

**Comment by panglesd** [comment]
Two comments:
- Could you rebase with the latest changes on `main`? You are using a pretty old version!
- It would be better to use the Notty attributes to colorize rather than the Ansi library. Feel free to take the newest changes as inspiration!

**Comment by ademolaomosanya** [comment]
> Two comments:
> 
> * Could you rebase with the latest changes on `main`? You are using a pretty old version!
> * It would be better to use the Notty attributes to colorize rather than the Ansi library. Feel free to take the newest changes as inspiration!

okay thank you 
i will do that

File: lib/interactive_viewer.ml [hunk 1]
```diff
@@ -159,20 +164,4 @@ let view (patches : Patch.t list) =
...
-                     `Handled
-                 | _ -> `Unhandled)
-               (W.string
-                  "Type 'h' to go to the help panel, 'q' to quit, 'n' to go to \
-                   the next operation, 'p' to go to the previous operation");
-        ]
-  in
-  W.vbox [ ui ]
-
-let start patch = Ui_loop.run ~quit ~tick_period:0.2 (view patch)
```
[the remark this answers is not in the document]
Comment (ademolaomosanya) [reply lib/interactive_viewer.ml:178]: > Why are all this removed? I guess it is a mistake from the rebasing?

yeah it must be a mistake

File: lib/interactive_viewer.ml [hunk 2]
```diff
@@ -47,7 +47,12 @@ let ui_of_operation operation =
       Ui.hcat
         [ W.string "Modification of "; W.string ~attr:blue_bold_attr path ]
 
-let string_of_hunk = Format.asprintf "%a" Patch.pp_hunk
+let colorize_hunk hunk =
+  let open Ansi in
```
[the remark this answers is not in the document]
Comment (ademolaomosanya) [reply lib/interactive_viewer.ml:51]: > Let's use what we have in `Nottui` and not the `Ansi` module.

okay

File: lib/interactive_viewer.ml [hunk 3]
```diff
@@ -47,7 +47,12 @@ let ui_of_operation operation =
...
+  match hunk with
+  | Patch.Context lines -> sprintf "%s%s%s" (fg_color Yellow) lines reset_color
+  | Patch.Addition lines -> sprintf "%s%s%s" (fg_color Green) lines reset_color
+  | Patch.Deletion lines -> sprintf "%s%s%s" (fg_color Red) lines reset_color
```
[the remark this answers is not in the document]
Comment (ademolaomosanya) [reply lib/interactive_viewer.ml:55]: > Additions and deletion should still show a + and - sign!

okay

**Comment by yokurang** [comment]
No progress for a while. I suggest closing the PR @Julow / @panglesd ?

**Comment by yokurang** [comment]
Already fixed in PR https://github.com/panglesd/diffcessible/pull/57

**Comment by panglesd** [comment]
Thanks anyway @ademolaomosanya for your work on this!

**Pull request closed without merging**